sources/packages/api/src-sql/Applications/Add-col-program-year-id.sql

```diff
@@ -0,0 +1,9 @@
+-- Create program_year_id
+ALTER TABLE
+    sims.applications
+ADD
+    COLUMN IF NOT EXISTS program_year_id INT REFERENCES sims.program_years(id) ON DELETE
+SET
+    NULL;
+
+COMMENT ON COLUMN sims.applications.program_year_id IS 'References the program year related to the application, this will be saved only when the ProgramYear is active and the application is Submitted and not in Draft state';
```

Comment:
Draft applications need a program year associated with. Since the draft application is also coming from a Form.IO definition, that came from a program year, we should be saving the program year also during the draft application.

Reply:
I dont think, we need to calculate the program year in draft state, we can discuss in a call to confirm this. As per discussion with @abschwenker , we came up with a requirement to save it while its submitted. @abschwenker Please confirm

Reply:
@guru-aot  not sure if you had a look at how the draft applications now work, but a "save draft" is now a create/update in our db so there is some core information that will be be required. A draft submission should have the PY saved with it because once created it must match without the possibility to be matched to another PY

Reply:
Yeah, when we submit the draft, the Program Year validation will happen, but the question is do we need to have the PY saved while the application is saved in draft state. Because last time, when we had a discussion, we agreed upon saving the PY only on successful submission and validation.

Reply:
yes we should. There will also be draft cleanup tasks we have to do and it would be really easy if the PY was saved with draft to move all drafts belonging to a PY

Reply:
Okay then if we agree on having the PY in Draft state saving, I can change my column to not null. For now, I have done my changes for the Submitted state. For Draft state integration, it has to be taken care. Fair enough? @abschwenker @andrewsignori-aot ?